app/components/build/AvatarWithUnknownEmailPrompt.js

```diff
@@ -60,19 +62,36 @@ class AvatarWithUnknownEmailPrompt extends React.Component {
     }
   }
 
+  // Given an email address, returns true/false depending on whether or not the
+  // current user owns that email and it's been verified
   isCurrentUsersValidatedEmail(email) {
-    const userEmail = this.getUserEmail(email);
+    const userEmail = this.findUserEmailNode(email);
+
     return userEmail && userEmail.verified;
   }
 
+  // Determines if this paticular build should prompt the current user to add
+  // it's associated email to their user account
+  isUnregisteredWithEmail(createdBy) {
```

Comment:
I just extracted this logic out to a method just to clean up the `componentDidMount` gear a little bit